packages/headless/src/modules/super-admin/service/index.ts

```diff
@@ -1,59 +1,295 @@
+      : await this.sendMfaOtp(id, { phone, password });
+    return sendOtp;
+  }
+
+  async superAdminLogin(
+    body: SuperAdminLoginDto,
+  ): Promise<IServiceResponse<SuperAdminLoginRes | OtpResponseDto>> {
+    const userData = await this.superAdminRepository.findOne({
+      email: body.email,
+    });
+    if (!userData) {
+      return errorResponse(
+        'Invalid credentials!',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const matchPassword = await bcrypt.compare(
+      body.password,
+      userData.password,
+    );
+    if (!matchPassword) {
+      return errorResponse(
+        'Invalid credentials!',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    if (userData.isMfaEnabled) {
+      userData.password = body.password;
+      const res = await this.handleMfaLogin(userData);
+      return res;
+    }
+
+    const payload: AdminJwtPayload = {
+      id: userData.id,
+      username: userData.firstName + ' ' + userData.lastName,
+      logInTime: Date.now(),
+      role: userData.role,
+    };
+    const token = this.jwtService.sign(payload);
+    return successResponse(SuperAdminLoginRes, { token });
+  }
+
+  async verifyMfaLoginOtp(
+    body: MfaVerifyOtpDto,
+  ): Promise<IServiceResponse<SuperAdminLoginRes>> {
+    if (!body.email && !body.phone) {
+      return errorResponse(
+        'Require email or phone',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    let query: Record<string, any> = {
+      isVerified: false,
+      id: body.id,
+      otp: body.otp,
+      otpExpireTime: { $gt: Date.now() },
+    };
+
+    if ((body.email && !body.phone) || (body.email && body.phone)) {
+      query.email = body.email;
+    } else {
+      query.phone = body.phone;
+    }
+
+    const verifiedData = await this.superAdminRepository.verifyOtp(query);
+    if (verifiedData) {
+      await this.superAdminRepository.deleteOtp({
+        ...query,
+        isVerified: true,
+      });
+      const userData = await this.superAdminRepository.findOne({
+        id: verifiedData.userId,
+      });
+      const payload: AdminJwtPayload = {
+        id: userData.id,
+        username: userData.firstName + ' ' + userData.lastName,
+        logInTime: Date.now(),
+        role: userData.role,
+      };
+      const token = this.jwtService.sign(payload);
+      return successResponse(SuperAdminLoginRes, { token });
+    } else {
+      return errorResponse('OTP is not verified', null, HttpStatus.CONFLICT);
+    }
+  }
+
+  //--- Multi factor auth functionalities---
+
+  async sendMfaOtp(
+    userId: string,
+    body: MfaOtpDto,
+  ): Promise<IServiceResponse<OtpResponseDto>> {
+    if (!body.email && !body.phone) {
+      return errorResponse(
+        'Require email or phone',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const isPasswordMatched = await this.superAdminHelperService.checkPassword(
+      body,
+      userId,
+    );
```

Comment:
why we are validating the password here?
two problem found so far.
- already the caller function is checking the password
- the function name does not match with its functionality

Reply:
1) Currently we are checking password at the time of login only.
2) To setup mfa, we should verify the real user. So, this function takes email or phone to send otp as well as takes password in same request body to verify the password further.